### PR TITLE
Add support for DreamGen API.

### PR DESCRIPTION
**Pull request by DreamGenX**
API docs: https://dreamgen.com/docs/models/opus/v1
API keys: https://dreamgen.com/account/api-keys

I decided to base this on the text-completion API since it's more flexible with SillyTavern's prompt formating capabilities.

This also means that custom context and instruct settings are required.

Will add documentation in a followup PR.

### CODE REVIEW

File: public/scripts/textgen-models.js [hunk 1]
```diff
@@ -327,10 +375,25 @@ export function getCurrentOpenRouterModelTokenizer() {
     }
 }
 
+export function getCurrentDreamGenModelTokenizer() {
+    const modelId = textgen_settings.openrouter_model;
```
Comment (Cohee1207) [inline public/scripts/textgen-models.js]: `openrouter_model` is likely a mistake

File: public/scripts/textgen-models.js [hunk 2]
```diff
@@ -327,10 +375,25 @@ export function getCurrentOpenRouterModelTokenizer() {
...
+    const model = openRouterModels.find(x => x.id === modelId);
```
Comment (Cohee1207) [inline public/scripts/textgen-models.js]: And here too. `openRouterModels`

File: public/index.html [hunk 3]
```diff
@@ -1276,6 +1276,11 @@ <h4 class="wide100p textAlignCenter"><span data-i18n="GBNF Grammar">GBNF Grammar
                                         <input class="neo-range-slider" type="range" id="min_length_textgenerationwebui" name="volume" min="0" max="2000" step="1" />
                                         <input class="neo-range-input" type="number" min="0" max="2000" step="1" data-for="min_length_textgenerationwebui" id="min_length_counter_textgenerationwebui">
                                     </div>
+                                    <div data-newbie-hidden data-tg-type="dreamgen" class="alignitemscenter flex-container flexFlowColumn flexBasis48p flexGrow flexShrink gap0">
+                                        <small data-i18n="Min Message Length">Min Message Length</small>
```
Comment (Cohee1207) [inline public/index.html]: Why not just reuse the existing min length control? Just send it to the API as an alias

File: public/scripts/textgen-models.js (hunk 1 above)
Comment (DreamGenX) [reply public/scripts/textgen-models.js]: Oppsie! Fixed.

File: public/scripts/textgen-models.js (hunk 2 above)
Comment (DreamGenX) [reply public/scripts/textgen-models.js]: Fixed.

File: public/index.html (hunk 3 above)
Comment (DreamGenX) [reply public/index.html]: Good idea, that works.

**Comment by DreamGenX** [comment]
Thank you for the reviewing the change and pointing the issues!

**Pull request merged**